src/model.rs

```diff
@@ -174,6 +171,16 @@ pub struct DataProduct {
 }
 
 impl DataProduct {
+    /// Pull a Data Product
+    pub async fn from_db(
```

Comment:
@coderabbitai Please add a unit test where this function can read a data product record from the DB, and a unit test where there data product does not exists.

Reply:
Please add a unit test where this function can read a data product record from the DB, and a unit test where there data product does not exists.

Reply:
Please create a PR for this change.